mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -491,6 +494,68 @@ private void appendJoinAlternative(SelectJoinStep<?> selectJoinStep, ParseLeader
     }
   }
 
+  @Override
+  public Future<RecordsIdentifiersCollection> getMatchedRecordsIdentifiers(MatchField matchedField, TypeConnection typeConnection,
+                                                                           boolean externalIdRequired, int offset, int limit, String tenantId) {
+    Table<org.jooq.Record> marcIndexersPartitionTable = table(name(MARC_INDEXERS_PARTITION_PREFIX + matchedField.getTag()));
+    if (matchedField.getValue() instanceof MissingValue) {
+      return Future.succeededFuture(new RecordsIdentifiersCollection().withTotalRecords(0));
+    }
+
+    return getQueryExecutor(tenantId).transaction(txQE -> txQE.query(dsl -> {
```

Comment:
Are there facilities to disable returning a count as part of a API response. If so, we should make the count call to be optional.

Reply:
Also, when mod-inventory makes a call to do matching, it should exclude the count.

Reply:
It is possible not to return totalRecords field in a response by setting it to null. Assume that if it's necessary, the logic to count and return "totalRecords" field can be managed by specifying "returnTotalRecords" query parameter in a request that would be true by default.


> Also, when mod-inventory makes a call to do matching, it should exclude the count.

Why mod-inventory should exclude the total records count? How would mod-inventory know that it has received identifiers for all records that match the matching criteria and doesn't need to fetch next batch? As far as understand mod-inventory should get identifiers of all matched records when such records' identifiers are needed for subsequent processing of instances matching.

Reply:
> Will the current state of the code handle this?

The current state will not handle it. I just described a possible approach how it can be implemented. Are you ok with the described approach for making total count calculation optional?

> That is a good point about next batch. The main reason to have the total count configurable is for performance. I assume that matching via kafka messages didn't need to make a call to get total count of the database. I think this shift to REST api to do matching could cause performance degradation due to extra calls to support the CQL/FOLIO HTTP response standard. There should be some way to know that there are more batch, that may not play with FOLIO for now.

Yes, the previous marc-marc matching approach did not need to get the total count because multiple matched records were treated as error. During marc-marc matching feature discussion we talked that multiple results from marc matching will be used in case of subsequent processing of instances matching. I'm thinking maybe mod-inventory could get total records count only by first request to records matching API, and for subsequent requests mod-inventory could specify query parameter not to perform total records calculation.

Reply:
changed the logic for total records calculation to optional but by introducing "returnTotalRecordsCount" field in the request body instead of "returnTotalRecords" query parameter as I described above.